The reverse wallening, from keep to manor

FieryTech opened the pull request:
## About The Pull Request

<!-- Describe your pull request. Avoid text walls, use concise bullet points for easier readability. Document every change, or this can delay review and even discourage maintainers from merging your PR. -->

Z1:
<details> 

<img width="1504" height="2016" alt="StrongDMM-2025-08-17 12 53 57" src="https://github.com/user-attachments/assets/c98bd425-8c81-4e6a-aaa1-09abe70d5c6f" />

</details>

Z2:
<details> 

<img width="1504" height="2272" alt="StrongDMM-2025-08-17 12 54 07" src="https://github.com/user-attachments/assets/3ff596aa-5595-4674-a9d6-1927718bc0a6" />

</details>

Z3:
<details> 

<img width="1536" height="2336" alt="StrongDMM-2025-08-17 12 54 15" src="https://github.com/user-attachments/assets/4968573a-1bf0-46ef-892e-12d03207c6a2" />

</details>

Z4:
<details> 

<img width="1504" height="2368" alt="StrongDMM-2025-08-17 12 54 22" src="https://github.com/user-attachments/assets/66d9e476-b0fa-40ca-b796-5760ff6bf751" />

</details>


## Testing Evidence

<!-- It's mandatory to test your PR. Provide images, clips or description of how you tested your changes where possible. -->

Don't you trust me?

<img width="703" height="593" alt="Capture d'écran 2025-08-17 101523" src="https://github.com/user-attachments/assets/68aa3f77-7e1f-43f2-9fc9-c16c8a1bd132" />


## Why It's Good For The Game

<!-- Argue for the merits of your changes and how they benefit the game, especially if they are controversial. If you can't, then it probably isn't good for the game in the first place. -->

We have been talking about this for months now: The keep as a whole is detrimental to the roleplay environment of the town. It's massive, block navigation, prevents people from seeing the nobles, knights, and such without having to yell over SCOM.

This project:
- Make the mages a bit more independant by housing them in the University of Azuria, alongside the archivist.
- Move the manor north, and add several layers of protection (including indestructible walls and unclimbable walls).
- Add the Azure Justiciar. Finally, the marshall and captain will have their own court house.
- Prepare room for future expansions by leaving the old library in the map, and empty spaces in waiting.

<!-- By contributing to this codebase, you confirm that any code and sprites you provide are legal to share and will be licensed under the terms specified in README.md — AGPLv3 for code and CC-BY-SA 3.0 for assets, unless otherwise stated. You acknowledge that the project maintainers are under no obligation to remove any materials that do not violate these licenses. -->

moltosotto commented:
Mages continue to get the coolest things... 😢

Great PR, though.

Ketrai commented:
Honestly with how the church vault isn't an issue and theft is so, so very hard to avoid. This is a monumental improvement. The keep should be hard to break into has never worked as a mentality and only served to be a detriment to roleplay.

That being said I do sorta worry a little about the defensibility of the keep & town as a whole against undead armies... but maybe that's something to solve with balancing rather than making every noble ever invisible.

FieryTech commented:
> Honestly with how the church vault isn't an issue and theft is so, so very hard to avoid. This is a monumental improvement. The keep should be hard to break into has never worked as a mentality and only served to be a detriment to roleplay.
> 
> That being said I do sorta worry a little about the defensibility of the keep & town as a whole against undead armies... but maybe that's something to solve with balancing rather than making every noble ever invisible.

Now that the main garrison is outside of the keep, with an armory that easily can be deployed when the manor is occupied by an invader, I do believe that taking it back might be slightly easier. The church also is another easily defensible position when that happens. The manor also can be assaulted from the north and south without the gigantic walls.

That is my personal opinion, but I think losing can be 'fun', and that they always have a way to come back with that.

Ketrai commented:
> > Honestly with how the church vault isn't an issue and theft is so, so very hard to avoid. This is a monumental improvement. The keep should be hard to break into has never worked as a mentality and only served to be a detriment to roleplay.
> > That being said I do sorta worry a little about the defensibility of the keep & town as a whole against undead armies... but maybe that's something to solve with balancing rather than making every noble ever invisible.
> 
> Now that the main garrison is outside of the keep, with an armory that easily can be deployed when the manor is occupied by an invader, I do believe that taking it back might be slightly easier. The church also is another easily defensible position when that happens. The manor also can be assaulted from the north and south without the gigantic walls.
> 
> That is my personal opinion, but I think losing can be 'fun', and that they always have a way to come back with that.

I wouldn't say church is very defensible given it's got a million breakable ground level windows. Losing is mostly fun when it's not a total wipeout. But with keep being less of an inescapable deathtrap, running from a losing battle should be more of an option again.

Ollanius commented:
The University-and-Justiciary of Azuria are very lovely additions.

I really, _really_ don't think that removing the walls from the Keep is a good idea, however.
From what I've personally seen, the Keep gets crapped on a _lot_ more than it should - changing it to be more _"roleplay-friendly"_ would require the roles _and_ the players to be tweaked, rather than the map itself.

As of right now, the Keep is the _only_ well-defended structure in the Town. It's _meant_ to be a fortress _(like an actual, historical castle)_ that can serve as the _"last bastion"_ in the event of a siege. Stripping out a majority of its defenses will make it _far_ easier  for antagonists to assault-and-usurp the Duke's throne, and for random people _(who REALLY shouldn't be in the Keep)_ to simply waltz-or-break in.

The indestructible hedges are a bit funny, but I don't think we should do that. It doesn't make much sense.

FieryTech commented:
On the first point, I always am happy to implement the suggestions people might have that would increase the manor's security. But as of now: The fences are reinforced to have stone-wall tier integrity, the area still has a moat and bridge that can be blocked, many places have been made impossible to climb, Ironically, removing the walls also makes me able to manage people jumping onto roofs way easily. I'm not saying it's as good as it used to be, but it still should be a pain to assault for any attacker.

The garrison also now has it's own fort north of town, with another armory, and they can access the manor way faster in case of emergency now that the north is open to them. Distances are important, and making the manor integrated into the town's tissue will allow for better manoeuvers and counter attacks. -> Plus, you also have the various adventurer vigilantes who can run after the antags in the area too.

If we speak historically too, many large cities had residential castles. Having the throne physically won't do much for you either if the other factions (church, mages, crafters, merchant, etc) aren't on board with you.

For the last paragraph, I assure you that most of the changes proposed here have been discussed publicly with the devs on the discord, and players I know are active in the garrison (And I play plenty of keep too, mind you). I say, let us try and see how people react. If it really is as bad as people make it seem, then Free will remove it as fast as it was pushed for test.

What is certain is that it will be maintained and changed, I am not assuming this solution is perfect.

ThinkArbies commented:
> As of right now, the Keep is the _only_ well-defended structure in the Town. It's _meant_ to be a fortress _(like an actual, historical castle)_ that can serve as the _"last bastion"_ in the event of a siege. Stripping out a majority of its defenses will make it _far_ easier for antagonists to assault-and-usurp the Duke's throne, and for random people _(who REALLY shouldn't be in the Keep)_ to simply waltz-or-break in.

Honestly, I think making it a little more open to attacks could be a good thing towards fighting the eternal Nothing Ever Happens:tm:. I'd like to see how it plays out before we completely throw that baby out with the bathwater.

The obvious downside, though, is making thievery far too easy. The "thief" gameplay loop is problematic and isn't conducive to RP. No guard wants to be on an eternal patrol path, and it's too cheap and easy otherwise to just slip in through the open gate, climb a wall and start raiding the duke's apartment.

MsCroakie commented:
I am really liking how open things are.

As a mostly towner player, it feels like the town has doubled in size, which I really like. The loss of the really cool gate is kinda sad, but the college and the courthouse are the coolest additions I've seen yet.

The only thing I'm sad to see so far, is losing the original library's very cool basement. I really enjoyed the moody blue lighting and downstairs portion, and I wish that could be retained and carried over to the college (though i am not sure there's space for it with all the basement stuff.)
<img width="284" height="294" alt="image" src="https://github.com/user-attachments/assets/f9c695a0-d5e9-4798-a029-a0fb9cdf55e1" />
Maybe it wouldn't make sense from an overhead perspective, but maybe a little library basement area could be fit here to retain the very lovely basement that no longer exists, or at least something reminiscent of it?

This is just a silly request from someone who's put a lot of rounds into Archivist, and that was my favorite area.

Mojamble commented:
Hi! My only complaint with this is the hollow shell that got left in the Archivist's old area. Otherwise? Peak. I'd personally say try to provide the manor with a LITTLE more security, but otherwise, it's cool.

FieryTech commented:
@MortoSasye It's solved.
@MsCroakie I can't easily place a basement. As a stop-gap, I switched the candles of the private room to blue.
@Mojamble It's a placeholder until we find something to place there. So many projects at once...

@radarseas We are going to improve it. I already got a good few feedbacks from the first few rounds in the new keep, and that prompted me to redo the entire manor's face with the help of a few keep players. As for the adventurer swam and such, I do believe it is the project of the coders to make communications a bit more difficult (you can see their PR!)

FieryTech commented:
Alright people, time for patch 2. It adds:
- Stone walls north and west of the manor.
- The old library has been replaced by an abandoned shop.
- A shopping list of small adjustments.

@ToasterStrudes
- The door was supposed to be a decoration. Removed it so people don't get confused.
- For the cat, I'll think about it, and see what we can do about it.
- For the peddler, you need to use the key, not your entire keyring. That's a known bug.

@squidqueen
- I'm not sure what you are talking about, I do not see any chains around the duke's bedroom. 
- Alright, I'll probably add a few rous under Kybraxor.
- No, we won't add back the large 3-tiles thick walls. I just added more defenses again.

ThinkArbies commented:
Honestly? I'm team open keep.

Sure, some tweaks will need to be made to make it less penetrable from the sides. I honestly don't think the keep should _also_  have the same fenceline on the north side, even if it does open up travel a bit more. And it's kind of a shame to see the court mage lose their cool aesthetic tower to turn them into a glorified college professor. The walls along the fenceline are two z-levels high which is nice, but the greenhouse nullifies that by being easier to climb than if the walls were just a simple brick wall anyways.

But overall? Good PR. The court feels more lively and easier to approach and request an audience with, and the rest of the keep grounds actually feels like a well-developed part of town, as opposed to the poopenfarten patchwork cobblestone of the rest of the town. The fact that it's a little more open to siege actually encourages sieges to happen (and also keeps occupying forces from holding a hellchoke at the gatehouse that just results in being shot endlessly by archers)
